src/log.js

```diff
@@ -676,3 +676,74 @@ export function isFromEmbed(win, opt_element) {
   }
   return opt_element.ownerDocument.defaultView != win;
 }
+
+/**
+ * Throws an error if the first argument isn't trueish.
+ *
+ * Supports argument substitution into the message via %s placeholders.
+ *
+ * Throws an error object that has two extra properties:
+ * - associatedElement: This is the first element provided in the var args.
+ *   It can be used for improved display of error messages.
+ * - messageArray: The elements of the substituted message as non-stringified
+ *   elements in an array. When e.g. passed to console.error this yields
+ *   native displays of things like HTML elements.
+ *
+ * @param {T} shouldBeTrueish The value to assert. The assert fails if it does
+ *     not evaluate to true.
+ * @param {string=} opt_message The assertion message
+ * @param {*=} opt_1 Optional argument (Var arg as individual params for better
+ * @param {*=} opt_2 Optional argument inlining)
+ * @param {*=} opt_3 Optional argument
+ * @param {*=} opt_4 Optional argument
+ * @param {*=} opt_5 Optional argument
+ * @param {*=} opt_6 Optional argument
+ * @param {*=} opt_7 Optional argument
+ * @param {*=} opt_8 Optional argument
+ * @param {*=} opt_9 Optional argument
+ * @return {T} The value of shouldBeTrueish.
+ * @template T
+ * eslint "google-camelcase/google-camelcase": 0
+ */
+export function devAssert(shouldBeTrueish, opt_message, opt_1, opt_2,
+  opt_3, opt_4, opt_5, opt_6, opt_7, opt_8, opt_9) {
+  if (getMode().minified) {
+    return shouldBeTrueish;
+  }
+  return dev().assert(shouldBeTrueish, opt_message, opt_1, opt_2, opt_3,
+      opt_4, opt_5, opt_6, opt_7, opt_8, opt_9);
+}
+
+/**
+ * Throws an error if the first argument isn't trueish.
+ *
+ * Supports argument substitution into the message via %s placeholders.
+ *
+ * Throws an error object that has two extra properties:
+ * - associatedElement: This is the first element provided in the var args.
+ *   It can be used for improved display of error messages.
+ * - messageArray: The elements of the substituted message as non-stringified
+ *   elements in an array. When e.g. passed to console.error this yields
+ *   native displays of things like HTML elements.
+ *
+ * @param {T} shouldBeTrueish The value to assert. The assert fails if it does
+ *     not evaluate to true.
+ * @param {string=} opt_message The assertion message
+ * @param {*=} opt_1 Optional argument (Var arg as individual params for better
+ * @param {*=} opt_2 Optional argument inlining)
+ * @param {*=} opt_3 Optional argument
+ * @param {*=} opt_4 Optional argument
+ * @param {*=} opt_5 Optional argument
+ * @param {*=} opt_6 Optional argument
+ * @param {*=} opt_7 Optional argument
+ * @param {*=} opt_8 Optional argument
+ * @param {*=} opt_9 Optional argument
```

Comment:
why can't it be just variadic again?

Reply:
There is a comment. I was suspecting that the `apply` call may not be inlinable leading to small code size regression. Just an experiment for now/